source/server/server.cc

```diff
@@ -57,7 +57,7 @@ InstanceImpl::InstanceImpl(
     : init_manager_(init_manager), workers_started_(false), live_(false), shutdown_(false),
       options_(options), validation_context_(options_.allowUnknownStaticFields(),
                                              !options.rejectUnknownDynamicFields()),
-      time_source_(time_system), restarter_(restarter), start_time_(time(nullptr)),
+      time_source_(time_system), restarter_(restarter), start_time_(testSource()),
```

Comment:
For this change to work, the change will have to be quite a bit deeper. There was a longer discussion in another PR: https://github.com/envoyproxy/envoy/pull/7166 from @derekargueta . Please read the comments in that PR -- actually that PR got abandoned and closed so it might be a good starting point.

One thing has gotten easier since then: hot-restart -- I think that was the stumbling block for that PR -- has gotten simpler.

Reply:
I think the hot restart concerns are still the same - while we've evolved to using protobufs, the original concern was whether the C++ chrono time objects could be interpreted differently across hot restarts if a different stdlib (or version of stdlib) was used in the next hot restart.

> Also note the epoch in C++ chrono (both steady_clock, and system_clock) is not defined to a fixed time by the spec, so technically it may differs across standard libraries, so probably better to use system_clock::from_time_t and system_clock::to_time_t to do the conversion for hot restart RPC, as those are defined to the number of seconds elapsed since 00:00 hours, Jan 1, 1970 UTC.

https://github.com/envoyproxy/envoy/pull/7166#pullrequestreview-246892963

The `from_time_t`/`to_time_t` approach was added to that PR. I believe it was just short one code review nit, so yes it'd be a great place to start.